File: MongoDB/JavaODMs/odmstest/src/main/java/cz/cuni/mff/mongodb_java/morphia/models/tpc_h_relational/RegionR.java
package cz.cuni.mff.mongodb_java.morphia.models.tpc_h_relational;

import dev.morphia.annotations.Entity;
import dev.morphia.annotations.Id;

import java.util.ArrayList;

@Entity
public class RegionR {
    @Id
    //private ObjectId r_regionkey;
    private int r_regionkey;

    private String r_name;
    private String r_comment;

    // Morphia needs this no-arg constructor
    public RegionR() {}

    public RegionR(int r_regionkey, String r_name, String r_comment) {
        this.r_regionkey = r_regionkey;
        this.r_name = r_name;
        this.r_comment = r_comment;
    }
}

File: MongoDB/CSharpODMs/MongoDBEntities/MongoDBEntities/Models/TPC-H/RegionR.cs
using MongoDB.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace MongoDBEntities.Models.TPC_H
{
    public class RegionR : Entity
    {
       
        public int r_regionkey;
        
        public string r_name;
        public string r_comment;

        public RegionR(int r_regionkey, string r_name, string r_comment)
        {
            this.r_regionkey = r_regionkey;
            this.r_name = r_name;
            this.r_comment = r_comment;
        }
    }
}
